config/config.go

```diff
@@ -1121,6 +1129,35 @@ func (c *EC2SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+// OpenstackSDConfig is the configuration for OpenStack based service discovery.
+type OpenstackSDConfig struct {
+	IdentityEndpoint string         `yaml:"identity_endpoint"`
+	Username         string         `yaml:"username"`
+	UserID           string         `yaml:"userid"`
+	Password         string         `yaml:"password"`
+	TenantName       string         `yaml:"tenant_name"`
+	TenantID         string         `yaml:"tenant_id"`
```

Comment:
Nitpicking, but "project_id" is the current terminology - and one I would expect most users of OpenStack to know.